New tool: biopet-bamstats

rhpvorderman opened the pull request:
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).
***

This pull request was automatically generated by [sbt-bioconda](https://github.com/biopet/sbt-bioconda). A release of [biopet-bamstats](https://github.com/biopet/bamstats) is automatically generated if the `sbt biocondaRelease` command is issued. As part of this process the recipes are tested using CircleCi as described [here](https://bioconda.github.io/contribute-a-recipe.html#test-locally). 

***

### Tool description

BamStats is a package that contains tools
to generate stats from a BAM file,
merge those stats for multiple samples,
and validate the generated stats files.

     
#### Mode - Generate

Generate reports clipping stats, flag stats, insert size and mapping quality on a BAM file. It outputs
a JSON file, but can optionally also output in TSV format.

The output of the JSON file is organized in a sample - library - readgroup tree structure.
If readgroups in the BAM file are not annotated with sample (`SM`) and library (`LB`) tags
an error will be thrown.
This can be fixed by using `samtools addreplacerg` or `picard AddOrReplaceReadGroups`.
     
        

#### Mode - Merge

This module will merge bamstats files together and keep the sample/library/readgroup structure.
Values for the same readgroups will be added.
It will also validate the resulting file.
      
        

#### Mode - Validate

Validates a BamStats file.
If aggregation values can not be regenerated the file is considered corrupt.
This should only happen when the file has been manually edited.

For documentation and manuals visit our github.io page: https://biopet.github.io/bamstats

epruesse commented on recipes/biopet-bamstats/1.0/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  home: https://github.com/biopet/bamstats
+  license: MIT
+  summary: BamStats is a package that contains tools to generate stats from a BAM file, merge those stats for multiple samples, and validate the generated stats files.
+  description: "BamStats is a package that contains tools\nto generate stats from\
```
see below, use `description: |`

epruesse commented on recipes/biopet-bamstats/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  home: https://github.com/biopet/bamstats
+  license: MIT
+  summary: BamStats is a package that contains tools to generate stats from a BAM file, merge those stats for multiple samples, and validate the generated stats files.
+  description: "BamStats is a package that contains tools\nto generate stats from\
```
You can just write 
```
description: |
   Many
   lines
   of 
   text
```

epruesse commented on recipes/biopet-bamstats/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  - biopet-bamstats generate --help
+  - biopet-bamstats generate --version
+extra:
+  notes: 'biopet-bamstats is a Java program that comes with a custom wrapper shell
```
same here

epruesse commented on recipes/biopet-bamstats/1.0/build.sh:
```diff
@@ -0,0 +1,12 @@
+# https://github.com/bioconda/bioconda-recipes/blob/master/recipes/varscan/build.sh
+# This file was automatically generated by the sbt-bioconda plugin.
+
+outdir=$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM
```
version and build number are unnecessary. Conda will not install the same package in different versions into the same environment anyway.

rhpvorderman commented:
We use a third-party for the yaml parsing of course. (Snakeyaml). So did some investigation why it would parse the strings like this. The problem appeared to be spaces at line endings. This was fixed. Also added some newlines in the notes section (it was a long string previously).

Also updated the build script to only include the package name. Not the version or build number.

Thanks @epruesse for your comments.
Is the recipe up to standards now?

epruesse commented:
LGTM

The Markdown markup in your documentation will not show correctly on bioconda.github.io though. The generator expects ReStructuredText, but the text will be Jinja escaped before that anyway. 

See https://github.com/bioconda/bioconda-utils/blob/master/docs/source/templates/readme.rst_t

https://github.com/bioconda/bioconda-utils/blob/d06308a09a09a030ecb9e5fa16f98705718e2ab7/docs/source/generate_docs.py#L216-L230

If you're intereste - it might be worth adding some code to allow formatted package descriptions.
